scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/client/LoadOps.scala

```diff
@@ -119,6 +126,56 @@ final private[client] class LoadOps(client: Client, jobService: JobOps) {
       location = location
     )
 
+  /**
+   * Upload List of rows to Cloud Storage as Avro file and load to BigQuery table. Note that element
+   * type `T` must be annotated with [[BigQueryType]].
+   */
+  def uploadTypedRows[T <: HasAnnotation: TypeTag](
+    tableSpec: String,
+    rows: List[T],
+    tempLocation: String,
+    writeDisposition: WriteDisposition = WriteDisposition.WRITE_APPEND,
+    createDisposition: CreateDisposition = CreateDisposition.CREATE_IF_NEEDED
+  ): Try[TableReference] = {
+    val bqt = BigQueryType[T]
+
+    Try {
+      val out = new ByteArrayOutputStream()
+      val datumWriter = new GenericDatumWriter[GenericRecord]()
+      val dataFileWriter = new DataFileWriter[GenericRecord](datumWriter)
+      try {
+        dataFileWriter.create(bqt.avroSchema, out)
+        rows.foreach { row =>
+          dataFileWriter.append(bqt.toAvro(row))
+        }
+      } finally {
+        dataFileWriter.close()
+      }
+
+      val blobId =
+        BlobId.fromGsUtilUri(
+          s"${tempLocation.stripSuffix("/")}/upload_${RandomStringUtils.randomAlphanumeric(10)}.avro"
+        )
+      val blobInfo = BlobInfo.newBuilder(blobId).build
+      client.blobStorage.createFrom(
+        blobInfo,
+        new ByteArrayInputStream(out.toByteArray),
+        Storage.BlobWriteOption.doesNotExist(),
+        Storage.BlobWriteOption.crc32cMatch()
```

Comment:
what's this option needed for? 👀

Reply:
validate the checksum after upload using CRC32 :) why not?